### PR TITLE
add payments wrapper feature

### PR DESCRIPTION
**Pull request by psorinionut**

### CODE REVIEW

File: common/common_structs/src/wrapper_types.rs [hunk 1]
```diff
@@ -92,3 +94,50 @@ impl<M: ManagedTypeApi> RawResultWrapper<M> {
...
+        Self { payments }
+    }
+
+    pub fn push(&mut self, payment: EsdtTokenPayment<M>) {
+        if payment.amount > 0 {
+            self.payments.push(payment);
+        }
+    }
+
+    pub fn send_to_address<T: SendApi>(&self, address: &ManagedAddress<M>) {
```
Comment (dorin-iancu) [inline common/common_structs/src/wrapper_types.rs:122]: Remove `T` and just use `M`, it already has the `SendApi` bound.

File: common/common_structs/src/wrapper_types.rs [hunk 2]
```diff
@@ -92,3 +94,50 @@ impl<M: ManagedTypeApi> RawResultWrapper<M> {
...
+                        &payment.token_identifier,
+                        payment.token_nonce,
+                        &payment.amount,
+                        0,
+                        &ManagedBuffer::new(),
+                        &ManagedArgBuffer::new(),
+                    );
+                }
+            }
+        }
```
Comment (dorin-iancu) [inline common/common_structs/src/wrapper_types.rs:137]: Use multi-transfer here. Also, no need for `> 0` check, as you already do that when adding the payment.

File: common/common_structs/src/wrapper_types.rs [hunk 3]
```diff
@@ -92,3 +94,50 @@ impl<M: ManagedTypeApi> RawResultWrapper<M> {
...
+    }
+
+    pub fn to_vec(&mut self) -> PaymentsVec<M> {
+        return self.payments.clone();
```
Comment (dorin-iancu) [inline common/common_structs/src/wrapper_types.rs:141]:
```suggestion
    pub fn to_vec(self) -> PaymentsVec<M> {
        self.payments
```

File: common/common_structs/src/wrapper_types.rs (hunk 1 above)
Comment (psorinionut) [reply common/common_structs/src/wrapper_types.rs]: Updated.

File: common/common_structs/src/wrapper_types.rs (hunk 3 above)
Comment (psorinionut) [reply common/common_structs/src/wrapper_types.rs]: Updated.

File: common/common_structs/src/wrapper_types.rs [hunk 4]
```diff
@@ -92,3 +94,44 @@ impl<M: ManagedTypeApi> RawResultWrapper<M> {
...
+                address,
+                &self.payments,
+                0,
+                &ManagedBuffer::new(),
+                &ManagedArgBuffer::new(),
+            );
+        }
+    }
+
+    pub fn into_vec(&mut self) -> PaymentsVec<M> {
```
Comment (dorin-iancu) [inline common/common_structs/src/wrapper_types.rs:134]:
```suggestion
    pub fn into_vec(self) -> PaymentsVec<M> {
```

**Pull request closed without merging**